plugins/modules/api.py

```diff
@@ -303,7 +303,7 @@ def __init__(self):
             remove=dict(type='str'),
             update=dict(type='str'),
             cmd=dict(type='str'),
-            query=dict(type='str'),
+            query=dict(type='dict'),
```

Comment:
This is a backwards-compatibility breaking change. The old form needs to continue to work as well.

How about adding a new parameter for this, making the two mutually exclusive, and have some code that transforms the old `query` content to the new parameter?

Reply:
I think how to solve the same, something like "extended_query" ?
I'm still working on the code to add Or and In in the query, will push it when is ready for final review 
If everything look ok will fix the change log, docs, unites etc.

Reply:
As yml structure, code will find 'or:' and build the query , will be same for where(key.In) 
Also I'm adding additional checks if the whole dict is structure correctly 
```
        items:
          - .id
          - network
          - address
        where:
          - network:
             is: "192.168.58.0"
          - address:
             not: "192.168.58.254/24"
          - address:
             or: 
               - is: "192.168.58.1/24"
               - not: "192.168.58.2/24"
```

Reply:
actually will do both :) 
```
          - network:
             '==': "192.168.58.0"
          - network:
             'is': "192.168.58.0"
```

Reply:
push new code which have query and extended_query (code is still in progress) just for info!

Reply:
build_api_extended_query() is very bulky ..but work fine,  I will try to fix it when I finish the main functionalities with 'Or/In' queries

Reply:
extended_query code is ready for review ! 

extended_query  structure is:

```
extended_query:
    items:
      - ros_attribute1
      - ros_attribute2
      - ros_attribute3
    where:
      - ros_attribute1:
          operator: "value"
      - ros_attribute2:
          or:
           - operator1: "value1"
           - operator2: "value2"
      - ros_attribute3:
          in:
            - "value1"
            - "value2"
```
items: type(list) order is not important

'extended_query':can work only with 'items' (same as normal 'query') -> 'where' is not mandatory 

where: type(list) order is important for routeros api query result

Any 'where'  ros_attribute must be in 'items' !

where sub items:
ros_attribute: type(dict) single operator(key) / vlaue

operator:
'==' or 'is' 
'!=' or 'not'
'>' or 'more'
'<' or 'less'

'or': type(list) (value is same as ros_attribute except 'or', 'in') order is important - depend on all other query attribute, first match win 
'in': type(list) (value is a list) order is not important if any of ros_attribute is in 'in list': true


Example (ignore example values ):
```
extended_query:
  items:
    - network
    - address
  where:
    - network:
       is: "192.168.58.0"
    - address:
       less: "192.168.58.1/24"
    - address:
       or: 
         - more: "192.168.58.1/24"
         - '==': "192.168.58.2/24"
    - address:
       in:
         - "10.20.36.253/24"
         - "192.168.58.2/24"
```